**BacFluxL**
**A workflow for genomic analysis of bacterial long reads.**
                                                                                                               
BacFluxL v1.0.1

**Rationale**
The analysis of bacterial Whole Genome Sequencing (WGS) data necessitates the integration of various bioinformatics tools. BacFluxL is a follow-up version of BacFlux, specifically tailored for sequences generated by Oxford Nanopore Technologies. It is primarily intended for long reads sequenced with the latest chemistry R10.4.1 and basecalled with Dorado using either accurate or super accurate models.
BacFluxL allows researchers to concentrate on the biological implications of their data, minimizing the need for technical data analysis.

**Description**.
#Here's a breakdown of the BacFluxL workflow#:
1.	**Preprocessing of Long Reads**:
	ONT sequences are trimmed and quality filtered using [Filtlong](https://github.com/rrwick/Filtlong).

2.	**Assembly of Long Reads**:
	Filtered ONT reads are assembled using [Flye](https://github.com/fenderglass/Flye ). For more details, see the next step.

3.	**Decontamination and Error Correction**:
	Filtered reads are mapped back to contigs using [minimap2](https://github.com/lh3/minimap2 ) and [samtools](https://github.com/samtools/samtools ).
	Local alignments of contigs are performed against the [NCBI nt](https://ftp.ncbi.nlm.nih.gov/blast/db/ ) database using [BLAST+](https://blast.ncbi.nlm.nih.gov/doc/blast-help/).
	Contaminant contigs are checked with [BlobTools](https://github.com/DRL/blobtools) . Unless otherwise specified (see configuration section for more details), the output of this step will be parsed automatically to discard contaminants based on the relative taxonomic composition of the contigs.
	[Medaka](https://github.com/nanoporetech/medaka) can be used to generate a consensus sequence from the assembled contigs and the original long reads. This consensus sequence should have a higher accuracy than the original assembled contigs, but this is not always the case, especially if ONT reads were base-called with the latest versions of the super accurate model of [Dorado](https://github.com/nanoporetech/dorado). For a deeper insight, please refer to Ryan Wick's bioinformatics [blog](https://rrwick.github.io/2023/12/18/ont-only-accuracy-update.html ). By leaving the medaka_model parameter blank, the error correction step won't be performed. By providing a fast model as parameter, BacfluxL will set Flye to assemble long reads using the --nano-raw parameter. If either an accurate or super accurate model is provided, Flye will use the --nano-hq parameter, instead. Same will happen if the medaka_model is left blank, assuming that data are high quality.
	Bacterial chromosomes are reoriented using [dnaapler]( https://github.com/gbouras13/dnaapler), to start canonically with the dnaA sequence. Other replicons like plasmids and bacteriophages are also reoriented, using repA and terL, respectively, as starting point.
	Genome completeness and contamination of long-read assembled bacterial chromosomes are assessed with [CheckM](https://github.com/Ecogenomics/CheckM ) using taxon-specific markers.

4.**Taxonomic Analysis**:
	Accurate taxonomic placement is performed with [GTDB-Tk](https://github.com/Ecogenomics/GTDBTk) using a curated reference [database](https://gtdb.ecogenomic.org/).

5.	**Annotation**:
	Contigs are annotated using [Prokka](https://github.com/tseemann/prokka) and [Bakta](https://github.com/oschwengers/bakta) for functional prediction.
	Further functional annotation is provided with [EggNOG](https://github.com/eggnogdb).
	Secondary metabolites are inferred with [antiSMASH](https://github.com/antismash/antismash).

6.	**Antimicrobial Resistance (AMR)**:
	Contigs are screened for antimicrobial resistance and virulence genes using [ABRicate](https://github.com/tseemann/abricate).

7.	**Plasmids**:
	The presence of plasmids is investigated with [Platon](https://github.com/PlatONnetwork ) and confirmed with a BLAST search.

8.	**Prophages**:
	Contigs are screened for viral sequences with [VirSorter2](https://github.com/jiarong/VirSorter2 ), followed by [CheckV](https://bitbucket.org/berkeleylab/checkv/src/master/) for refinement.

9.	**Reporting**:
	Results are parsed and aggregated to generate a report using [MultiQC](https://github.com/MultiQC/MultiQC).

Installation
BacFluxL downloads automatically all dependencies and several databases. However, some external databases require manual download before running the workflow.
1.	**Download BacFluxL:**
Head over to the Releases section of the repository. Download the latest archive file (typically in .zip or .tar.gz format). This archive contains the BacFluxL Snakefile script, the config.yaml configuration file, and an envs environment directory. Extract the downloaded archive into your desired location. This will create a directory structure with the necessary files and directories. Alternatively, download via command line as:
#git command
git clone https://github.com/iLivius/BacFluxL.git
2.	**Install Snakemake:**
BacFluxL relies on Snakemake to manage the workflow execution. Find the official and complete set of instructions here. To install Snakemake as a Conda environment:
#install Snakemake in a new Conda environment
mamba create -c conda-forge -c bioconda -n snakemake snakemake
3.**Databases:**
While BacFluxL automates the installation of all software dependencies, some external databases need to be downloaded manually. Unless you have installed them already. In that case, skip this paragraph and jump to the configuration section.

Acknowledgements
This work was supported by the Austrian Science Fund (FWF) [Project I6030-B].
Citation
Antonielli, L., Koch, H., Trognitz, T., Nagel, M., Sanchez Mejia, A., & Großkinsky, D. K. (2024). BacFluxL: A workflow for genomic analysis of bacterial long reads. Zenodo. https://doi.org/10.5281/zenodo.11243909
References
1.	Blin, K., Shaw, S., Augustijn, H. E., Reitz, Z. L., Biermann, F., Alanjary, M., Fetter, A., Terlouw, B. R., Metcalf, W. W., Helfrich, E. J. N., van Wezel, G. P., Medema, M. H., & Weber, T. (2023). antiSMASH 7.0: New and improved predictions for detection, regulation, chemical structures and visualisation. Nucleic Acids Research, 51(W1), W46–W50. https://doi.org/10.1093/nar/gkad344
2.	Bouras, G., Grigson, S. R., Papudeshi, B., Mallawaarachchi, V., & Roach, M. J. (2024). Dnaapler: A tool to reorient circular microbial genomes. Journal of Open Source Software, 9(93), 5968. https://doi.org/10.21105/joss.05968
